BASE/src/ServerTelemetryChannel/Implementation/TelemetryBuffer.cs

```diff
@@ -203,6 +210,26 @@ IEnumerator IEnumerable.GetEnumerator()
             return this.GetEnumerator();
         }
 
+        private List<ITelemetry> MoveTelemeteryBufferForFlush()
+        {
+            List<ITelemetry> telemetryToFlush = null;
+            if (this.itemBuffer.Count > 0)
+            {
+                lock (this)
```

Comment:
I know this was in the original but, instead of `lock(this)`, we should use a dedicated syncObject for synchronization.